pkg/agent/baker.go

```diff
@@ -161,6 +161,17 @@ func (t *TemplateGenerator) getSingleLine(textFilename string, profile interface
 	if err != nil {
 		return "", fmt.Errorf("yaml file %s does not exist", textFilename)
 	}
+	lines := strings.Split(string(b), "\n")
+	contentToKeep := []string{}
+	for _, line := range lines {
+		trimmedToCheck := strings.TrimSpace(line)
+		if strings.HasPrefix(trimmedToCheck, "# ") || strings.HasPrefix(trimmedToCheck, "##") {
```

Comment:
can we just check if `strings.HasPrefix(trimmedToCheck, "#")`?

I'd also refactor to a separate func if possible

Reply:
Some bash lines start with "#", for example `#!/bin/bash`
Can move to separate func

Reply:
are there others that start with the same? maybe we can just exclude that?

open to either approach though, I see what you're saying...

Reply:
Not sure honestly, but I think there are others, for example `#>` that appears in custom data